### PR TITLE
Exit with code 1 on errors instead of 0

### PR DESCRIPTION
**Pull request by BodilessSleeper**
Fixes issue #38

### CODE REVIEW

**Comment by lucascebertin** [comment]
Hey, thanks for your contribution, we are trying to address this in a different way...
Please refer to this [pull request](https://github.com/nopsteam/6502/pull/40) and feel free to comment there.
Maybe we can close this if we agree to the approach.

**Comment by BodilessSleeper** [comment]
To be honest, if you're going to write out the reason you exited with 1 from your program you don't really need to return anything and check for it later only to quit out anyways, but let's be real you're only wasting a couple of CPU cycles here and there on returning and checking if the return is good or not (also, you should probably fprintf to stderr when you encounter an error, otherwise those printfs are treated as stdout and if someone were to redirect only legit output without errors he'd get errors anyways because printf only prints to stdout and fprintf can print to any file stream you specify it to, whether it's a file or a file desctiptor like stderr)

**Comment by rafaeldelboni** [comment]
Hey, @BodilessSleeper thanks for your inputs, we (me and @lucascebertin) are both very noobs with pure C, we wanted to test all the behaviors of the returns with an error, so if we want to test the behavior of `exit(1)` is that possible?

Thanks for your comment I didn´t knew about the differences in the printf´s.

**Comment by BodilessSleeper** [comment]
I'm not an expert on what you guys are doing either, so I'm not really sure how you should really tackle the problem and what should you really check for so (I also got no idea how unit tests on github work, I can debug my own code, and test it too, but unit tests are just a bit of a strange concept to me)
I am gonna tell you what I do know, namely, exit(some_code) is gonna terminate the whole process immediately with the code you specify, but it is returned to the system, not a function, while return(some_code) will terminate the function and return the specified code to the place where the function was called and you can check what that returned
Basically if you can you can write tests to check system return codes and stderr output you're better off with fprintf(stderr, "error_text"); exit(1); and test for exit code 1 (meaning you errored out), and the error text for a specific function since getting 1 can mean lots of things (eg. test for "cannot open file", when testing for check_if_cant_read_file), otherwise if you can only check function return codes you should just fprintf(stderr, "error_text"); return 1; and check only a specific function for return value (no need to check the error text when you only check a specific function)

**Comment by lucascebertin** [comment]
So to understand a better way to go we need to understand if our test suite can check for error codes returned by an exit(x).
I say that because the test suite is building binaries that runs and asserts the code.
But let's figure it out..

**Comment by rafaeldelboni** [comment]
Yeah, testing functions with exit(something) is very hard, you need to change the code to be able to mock the exit function
https://locklessinc.com/articles/mocking/
Or use something like the death checks like the google test suite has [here](https://github.com/google/googletest/blob/daa0df7b410cfed62ef474446eff22148a272a8d/googletest/include/gtest/gtest-death-test.h#L83), but our suite (Unity) doesn´t have something like this as I searched in the docs. 

But I changed the printf as you suggested to fprintf to stderr and added you [as the co-author](https://github.com/nopsteam/6502/pull/40/commits/3672dd9144115087fe98b9f206c22535812b4fca) @BodilessSleeper 

Thanks a lot for your time and for sharing your knowledge.

**Comment by BodilessSleeper** [comment]
Yeah, I've been messing around for a bit and I realised what was going on (at least on the surface level, that the system builds the binary first and then the test binaries which are based on your .h files, meaning they can all have the same functionality, so checking for exit codes is a pain)
It isn't impossible to check for exit codes tho, as I'm pretty sure you could run your binary through the test binary (using things like system(), execve()), to run the whole binary that we just built, and check for return code of said functions (which will really just be your binary's exit code, but I'm not sure if stderr is readable that way, probably is though)
But again, your way of doing it is just way easier for unit testing and it wastes very small amount of CPU cycles
